subsys/settings/src/settings_line.c

```diff
@@ -87,3 +93,407 @@ int settings_line_make(char *dst, int dlen, const char *name, const char *value)
 
 	return off;
 }
+
+struct settings_io_cb_s {
+	int (*read_cb)(void *ctx, off_t off, char *buf, size_t *len);
+	int (*write_cb)(void *ctx, off_t off, char const *buf, size_t len);
+	size_t (*get_len_cb)(void *ctx);
+	u8_t rwbs;
+} settings_io_cb;
+
+#define MAX_ENC_BLOCK_SIZE 4
+
+int settings_line_write(const char *name, const char *value, size_t val_len,
+			off_t w_loc, void *cb_arg)
+{
+	size_t w_size, rem, add;
+
+#ifdef CONFIG_SETTINGS_USE_BASE64
+	/* minimal buffer for encoding base64 + EOL*/
+	char enc_buf[MAX_ENC_BLOCK_SIZE + 1];
+	char *p_enc;
+	size_t enc_len = 0;
+#endif
+
+	bool done;
+	char w_buf[16]; /* write buff, must be aligned either to minimal */
+			/* base64 encoding size and write-block-size */
+	int rc;
+	u8_t wbs = settings_io_cb.rwbs;
+#ifdef CONFIG_SETTINGS_ENCODE_LEN
+	u16_t len_field;
+#endif
+
+	rem = strlen(name);
+
+#ifdef CONFIG_SETTINGS_ENCODE_LEN
+	len_field = settings_line_len_calc(name, val_len);
+	memcpy(w_buf, &len_field, sizeof(len_field));
+	w_size = 0;
+
+
+	add = sizeof(len_field) % wbs;
+	if (add) {
+		w_size = wbs - add;
+		if (rem < w_size) {
+			w_size = rem;
+		}
+
+		memcpy(w_buf + sizeof(len_field), name, w_size);
+		name += w_size;
+		rem -= w_size;
+	}
+
+	w_size += sizeof(len_field);
+	if (w_size % wbs == 0) {
+		rc = settings_io_cb.write_cb(cb_arg, w_loc, w_buf, w_size);
+		if (rc) {
+			return -EIO;
+		}
+	}
+	/* The Alternative to condition above mean that `rem == 0` as `name` */
+	/* must have been consumed					     */
+#endif
+	w_size = rem - rem % wbs;
+	rem %= wbs;
+
+	rc = settings_io_cb.write_cb(cb_arg, w_loc, name, w_size);
+	w_loc += w_size;
+	name += w_size;
+	w_size = rem;
+
+	if (rem) {
+		memcpy(w_buf, name, rem);
+	}
+
+	w_buf[rem] = '=';
+	w_size++;
+
+	rem = val_len;
+	done = false;
+
+	while (1) {
+		while (w_size < sizeof(w_buf)) {
+#ifdef CONFIG_SETTINGS_USE_BASE64
+			if (enc_len) {
+				add = min(enc_len, sizeof(w_buf) - w_size);
+				memcpy(&w_buf[w_size], p_enc, add);
+				enc_len -= add;
+				w_size += add;
+				p_enc += add;
```

Comment:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/settings/src/settings_line.c:183:11: error: p_enc may be used uninitialized in this function [-Werror=maybe-uninitialized]